databroker/src/broker.rs

```diff
@@ -679,6 +719,28 @@ impl Subscriptions {
                 }
             }
         });
+
+        self.actuation_subscriptions.retain(|sub| {
+            if !sub.actuation_provider.is_available() {
+                info!("actuation provider gone: removing subscription");
+                false
+            } else {
+                match sub.permissions.expired() {
```

Comment:
I think this method has a somewhat strange name. Would it not be better to rename it to `is_valid()`, `valid()` or similar. Right now just by looking athe name I get the impression that it will return True/ok if it actually has expired.

Reply:
Yes, I had the same feeling while using the method. I think the method was written like this for easy reuse / easy return in the other methods e.g. permissions.rs#can_write_actuator_target()
```
    pub fn can_write_actuator_target(&self, path: &str) -> Result<(), PermissionError> {
        self.expired()?;

        if self.actuate.is_match(path) {
            return Ok(());
        }
        Err(PermissionError::Denied)
    }
```

The easiest way would probably be to simply return a bool in #expired() and simply adapt the other methods to a minimal more verbose version:
```
    pub fn can_write_actuator_target(&self, path: &str) -> Result<(), PermissionError> {
        if self.expired() {
            return Err(PermissionError::Expired);
        }

        if self.actuate.is_match(path) {
            return Ok(());
        }
        Err(PermissionError::Denied)
    }
```